src/PerformanceSensitiveAnalyzers/CSharp/CodeFixes/AvoidAllocationWithArrayEmptyCodeFix.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            return semanticModel.GetTypeInfo(objectCreationExpressionSyntax).Type is INamedTypeSymbol createdType &&
+                   (createdType.TypeKind == TypeKind.Array || IsCollectionType(createdType));
+        }
+
+        private static bool IsCollectionType(INamedTypeSymbol typeSymbol)
+        {
+            return typeSymbol.ConstructedFrom.Interfaces.Any(x =>
+                x.IsGenericType && x.ToString().StartsWith("System.Collections.Generic.ICollection", System.StringComparison.Ordinal));
+        }
+
+        private static bool IsPropertyTypeReadonlySequence(SemanticModel semanticModel, PropertyDeclarationSyntax propertyDeclaration)
+        {
+            return IsTypeReadonlySequence(semanticModel, propertyDeclaration.Type);
+        }
+
+        private static bool IsReturnTypeReadonlySequence(SemanticModel semanticModel, MethodDeclarationSyntax methodDeclarationSyntax)
+        {
+            var typeSyntax = methodDeclarationSyntax.ReturnType;
+            return IsTypeReadonlySequence(semanticModel, typeSyntax);
+        }
+
+        private static bool IsExpectedParameterReadonlySequence(SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (node is ArgumentSyntax argument && node.Parent is ArgumentListSyntax argumentList)
+            {
+                var argumentIndex = argumentList.Arguments.IndexOf(argument);
+                if (semanticModel.GetSymbolInfo(argumentList.Parent).Symbol is IMethodSymbol methodSymbol)
+                {
+                    if (methodSymbol.Parameters.Length > argumentIndex)
+                    {
+                        var parameterType = methodSymbol.Parameters[argumentIndex].Type;
+                        if (IsTypeReadonlySequence(semanticModel, parameterType))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static bool IsTypeReadonlySequence(SemanticModel semanticModel, TypeSyntax typeSyntax)
+        {
+            var returnType = ModelExtensions.GetTypeInfo(semanticModel, typeSyntax).Type;
+            return IsTypeReadonlySequence(semanticModel, returnType);
+        }
+
+        private static bool IsTypeReadonlySequence(SemanticModel semanticModel, ITypeSymbol type)
+        {
+            if (type.Kind == SymbolKind.ArrayType)
+            {
+                return true;
+            }
+
+            if (type is INamedTypeSymbol namedType && namedType.IsGenericType)
+            {
+                foreach (var readonlySequence in GetReadonlySequenceTypes(semanticModel))
+                {
+                    if (namedType.ConstructedFrom.Equals(readonlySequence))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
```

Comment:
What about the following syntax?

```suggestion
        => type.Kind == SymbolKind.ArrayType &&
            type is INamedTypeSymbol namedType && 
            namedType.IsGenericType &&
            GetReadonlySequenceTypes(semanticModel).Any(readonlySequence => namedType.ConstructedFrom.Equals(readonlySequence));
```